src/main/scala/com/example/akka/swagger/SwaggerDocService.scala

```diff
@@ -1,21 +1,102 @@
 package com.example.akka.swagger
 
-import com.github.swagger.akka.SwaggerHttpService
-import com.github.swagger.akka.model.Info
+import akka.http.javadsl.server.Rejections
+import akka.http.scaladsl.model.StatusCodes.{Found, PermanentRedirect, TemporaryRedirect}
+import akka.http.scaladsl.model.{ContentTypes, HttpEntity, StatusCodes}
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.directives.RouteDirectives.redirect
+import akka.util.ByteString
 import com.example.akka.add.AddService
 import com.example.akka.addoption.AddOptionService
 import com.example.akka.echoenum.EchoEnumService
 import com.example.akka.echoenumeratum.EchoEnumeratumService
 import com.example.akka.echolist.EchoListService
 import com.example.akka.hello.HelloService
+import com.example.akka.swagger.SwaggerDocService.swaggerUi
+import com.github.swagger.akka.SwaggerHttpService
+import com.github.swagger.akka.model.Info
 import io.swagger.v3.oas.models.ExternalDocumentation
+import org.webjars.{MultipleMatchesException, NotFoundException, WebJarAssetLocator}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
 
-object SwaggerDocService extends SwaggerHttpService {
+/**
+ * Webjars loaded/located via [[org.webjars.WebJarAssetLocator]] and served via akka-http.
+ */
+class SwaggerDocService(implicit val executionContext: ExecutionContext) extends SwaggerHttpService {
   override val apiClasses = Set(classOf[AddService], classOf[AddOptionService], classOf[HelloService],
     EchoEnumService.getClass, EchoEnumeratumService.getClass, EchoListService.getClass)
   override val host = "localhost:12345"
   override val info: Info = Info(version = "1.0")
   override val externalDocs: Option[ExternalDocumentation] = Some(new ExternalDocumentation().description("Core Docs").url("http://acme.com/docs"))
   //override val securitySchemeDefinitions = Map("basicAuth" -> new BasicAuthDefinition())
   override val unwantedDefinitions = Seq("Function1", "Function1RequestContextFutureRouteResult")
-}
\ No newline at end of file
+
+  val webJarAssetLocator = new WebJarAssetLocator()
+
+  /**
```

Comment:
Thanks very much for doing this. Looks good. I have one minor quibble. I can see how embedding the swagger-ui makes the demo easier to work with but I like the idea that users can use the sample as a template for their own app and they might not want to embed the UI. Would it be possible to move at least some of the web jar code to it's own trait or class (even if SwaggerDocService still calls out to this code)?

Reply:
Yes, I can imagine. I could create a SwaggerWithUIDocService that extends the SwaggerDocService. Then people can still use the SwaggerDocService if they don't want to include embed the swagger-ui.
I'll add a comment in the build.sbt on the webjar part too, so people know what to include or exclude.

Reply:
I've moved the SwaggerUI part to a trait, and split the dependencies in the build.sbt. So now it should be easy to use the example as template for application with or without the included UI.